unyt/array.py

```diff
@@ -166,6 +167,20 @@ def _iterable(obj):
     return True
 
 
+def _unit_operation_error_raise_or_warn(ufunc, u0, u1, func, *inputs):
```

Comment:
I am hesitant about how this function implements the proposed functionality. Specifically, it feels weird to replace a raise statement with a return statement (where this function is used).
Another comment is that the `func` argument is redundant (I think), since it can be obtained as `func = ufunc.method`.

Reply:
I share this uneasiness but I'm not sure of a better solution. It seems clear to me that some users will want this to fail explicitly and raise, whereas others will just want the warning, so there has to be a conditional somewhere that switches between that behaviour?

Reply:
To your second comment, makes sense, I'll remove the `func` argument.